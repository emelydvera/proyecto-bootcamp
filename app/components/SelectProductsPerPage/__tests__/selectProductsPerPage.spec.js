const React = require("react");
// import { act } from "react-dom/test-utils";
const SelectProductsPerPage = require("../SelectProductPerPage");
const { render, screen, fireEvent, act } = require("@testing-library/react");
const { getByRole, getAllByRole } = screen;
const UrlGenerator = require("../../../utils/urlGenerator");
const { I18nProvider } = require("nordic/i18n");
jest.mock("../../../utils/urlGenerator");

describe("SelectProductsPerPage component", () => {
  const i18n = { gettext: (text) => text };
  Object.defineProperty(window, "location", {
    value: {
      href: "http://localhost",
    },
    configurable: true,
  });

  beforeEach(() => {
    UrlGenerator.mockClear();
  });

  it("should render component", async () => {
    const urlGenerator = new UrlGenerator();

    const { asFragment } = render(
      <I18nProvider i18n={i18n}>
        <SelectProductsPerPage urlGenerator={urlGenerator} />
      </I18nProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should invoke urlGenerator.setQuery when clicking a dropdown item", () => {
    const urlGenerator = new UrlGenerator();

    render(
      <I18nProvider i18n={i18n}>
        <SelectProductsPerPage urlGenerator={urlGenerator} />
      </I18nProvider>
    );

    const dropdown = getByRole("button");
    fireEvent.click(dropdown);

    const dropdownItems = getAllByRole("option");
    fireEvent.click(dropdownItems[0]); // this is the first item of the dropdown containing value 5

    expect(urlGenerator.setQuery).toHaveBeenCalled();
    expect(urlGenerator.setQuery).toHaveBeenCalledWith("limit", 5);
  });

  it("should invoke urlGenerator.getQueryString when clicking a dropdown item", () => {
    const urlGenerator = new UrlGenerator();
    jest
      .spyOn(urlGenerator, "getQueryString")
      .mockImplementation(() => "q=mac");

    render(
      <I18nProvider i18n={i18n}>
        <SelectProductsPerPage urlGenerator={urlGenerator} />
      </I18nProvider>
    );

    const dropdown = getByRole("button");
    fireEvent.click(dropdown);

    const dropdownItems = getAllByRole("option");
    fireEvent.click(dropdownItems[0]); // this is the first item of the dropdown containing value 5

    expect(urlGenerator.getQueryString).toHaveBeenCalled();
    expect(window.location.href).toBe("/listado?q=mac");
  });
});
