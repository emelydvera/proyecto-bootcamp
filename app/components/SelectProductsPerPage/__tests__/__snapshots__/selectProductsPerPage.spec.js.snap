// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectProductsPerPage component should render component 1`] = `
<DocumentFragment>
  <div
    class="products__per-page"
  >
    <div
      class="andes-dropdown andes-dropdown--standalone products__dropdown andes-dropdown--compact"
    >
      <div
        class="andes-floating-menu"
      >
        <button
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-invalid="false"
          aria-label="Ordenar por: "
          class="andes-dropdown__trigger"
          type="button"
        >
          <span
            class="andes-dropdown__display-values"
          >
            Ordenar por
          </span>
          <svg
            aria-hidden="true"
            class="andes-dropdown__standalone-arrow"
            height="12"
            viewBox="0 0 12 12"
            width="12"
          >
            <path
              d="M6 7.057L9.352 3.705 10.148 4.5 6 8.648 1.852 4.5 2.648 3.705z"
              fill-opacity=".45"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
